v2/api-validator/tests/server-tests/bad-request-body.test.ts

```diff
@@ -41,6 +51,11 @@ describe('Test request bodies missing one required property', () => {
           beforeAll(async () => {
             const badBody = _.cloneDeep(goodBody);
             deleteDeepProperty(badBody, propertyPath);
+
+            if (badBody?.['idempotencyKey']) {
```

Comment:
`badBody` is never undefined here, so `?.` is redundant.

If I understand correctly your intention here, the semantic meaning of this condition is "if badBody has the property idempotencyKey, randomize its value". In this case, writing it as `if (badBody.hasOwnProperty('idempotencyKey'))` would convey better the meaning and be true even if the value of the property is undefined or null.